src/main/java/mr/limpios/smart_divide_backend/aplication/assemblers/ExpenseModelAssembler.java

```diff
@@ -0,0 +1,126 @@
+package mr.limpios.smart_divide_backend.aplication.assemblers;
+
+import mr.limpios.smart_divide_backend.domain.models.*;
+import mr.limpios.smart_divide_backend.domain.strategies.ParticipantsPayersOfExpenses;
+import mr.limpios.smart_divide_backend.domain.strategies.ValidatedMembers;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ExpenseModelAssembler {
+    public static List<ExpenseParticipant> createExpenseParticipantsFromValidatedParticipants(
+            ParticipantsPayersOfExpenses participantsPayersOfExpenses,
+            Map<String, User> groupMembersMap) {
+
+        Map<String, BigDecimal> payersMap = getMembersMap(participantsPayersOfExpenses.payers());
+        Map<String, BigDecimal> participantsMap = getMembersMap(participantsPayersOfExpenses.participants());
+
+        Set<String> allMemberIds = new HashSet<>();
+        allMemberIds.addAll(payersMap.keySet());
+        allMemberIds.addAll(participantsMap.keySet());
+
+        List<ExpenseParticipant> result = new ArrayList<>();
+        for (String memberId : allMemberIds) {
+            User user = groupMembersMap.get(memberId);
+            BigDecimal amountPaid = payersMap.getOrDefault(memberId, BigDecimal.ZERO);
+            BigDecimal mustPaid = participantsMap.getOrDefault(memberId, BigDecimal.ZERO);
+
+            result.add(new ExpenseParticipant(
+                    null,
+                    user,
+                    amountPaid,
+                    mustPaid
+            ));
+        }
+
+        return result;
+    }
+
+    private static Map<String, BigDecimal> getMembersMap(List<ValidatedMembers> members) {
+        return members.stream()
+                .collect(Collectors.toMap(
+                        ValidatedMembers::memberId,
+                        ValidatedMembers::amount
+                ));
+    }
+
+    private static List<String> getCreditors(Map<String, BigDecimal> balances) {
+        return balances.entrySet().stream()
+                .filter(e -> e.getValue().compareTo(BigDecimal.ZERO) > 0)
+                .map(Map.Entry::getKey)
+                .toList();
+    }
+
+    private static List<String> getDebtors(Map<String, BigDecimal> balances) {
+        return balances.entrySet().stream()
+                .filter(e -> e.getValue().compareTo(BigDecimal.ZERO) < 0)
+                .map(Map.Entry::getKey)
+                .toList();
+    }
```

Comment:
Puedes crear mejor un único método? Y pasarle un `predicado`, por ejemplo: 
```suggestion
       private static List<String> filterIds(Map<String, BigDecimal> balances, Predicate<BigDecimal> predicate) {
        return balances.entrySet().stream()
                .filter(e -> predicate.test(e.getValue()))
                .map(Map.Entry::getKey)
                .toList();
    }
```

Y usarlo así: 

```Java

   List<String> creditors = filterIds(balances, b -> b.compareTo(BigDecimal.ZERO) > 0);
        List<String> debtors   = filterIds(balances, b -> b.compareTo(BigDecimal.ZERO) < 0);

```